lib/ts/unit-tests/string_view.cpp

```diff
@@ -0,0 +1,516 @@
+/** @file
+  Test file for basic_string_view class
+  @section license License
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include "catch.hpp"
+
+#ifndef _DEBUG
+#define _DEBUG
+#endif
+
+#include "string_view.h"
+#include <iostream>
+#include <string>
+#include <vector>
+
+using namespace std;
+
+auto npos = ts::string_view::npos;
+
+// ======= test for string_view ========
+// test cases:
+//[constructor], [operator], [type], [access], [capacity], [modifier], [operation], [compare], [find]
+
+TEST_CASE("constructor calls", "[string_view], [constructor]")
+{
+  SECTION("Literal look for NULL")
+  {
+    ts::string_view sv("hello");
+    REQUIRE(sv.size() == 5);
+    REQUIRE(sv.length() == 5);
+    REQUIRE(sv.empty() == false);
+    REQUIRE(sv == "hello");
+  }
+
+  SECTION("operator =")
+  {
+    ts::string_view sv;
+    sv = "hello";
+    REQUIRE(sv.size() == 5);
+    REQUIRE(sv.length() == 5);
+    REQUIRE(sv.empty() == false);
+    REQUIRE(sv == "hello");
+  }
+
+  SECTION("Literal with NULL")
+  {
+    ts::string_view sv("hello\0world");
```

Comment:
Hmmmm. That's a philosophical difference from `StringView`. Give that literal the length would be 11. I've been thinking about removing the `const char*` constructor to avoid confusion on this point.